### PR TITLE
Turn exercise 011 into practice

### LINKED ISSUE #2179: Turn exercise 011 into practice
Extract the OCaml code from the markdown file corresponding to this exercise. The file is: `data/exercise/011_NAME.md` (where `011` is the number from the title of this issue and `NAME` is the rest of the file name, for instance `data/exercise/001_tail.md` is the file which corresponds to exercise `001`). Use the code in this file to create a folder `practice/011` (using this same number).

Populate the `011` folder with (almost) the same files as in folders `practice/001` and `practice/002` (they are examples).

Three files are different in each folder:
1. `practice/011/ex.ml`: it contains the function's signature, which needs to be implemented by the learner; 
1. `practice/011/work/impl.ml`: it includes the template meant to be filed by the learner;
1. `practice/011/answer/impl.ml`: consists of the solution from the original markdown file.

The examples included in the original markdown file must be turned into test cases in the `practice/011/ex.ml` file.

Folder structure to create:
```
practice/011/
├── answer
│   ├── impl.ml
│   └── test
│       ├── dune
│       └── run.ml
├── dune
├── dune-project
├── ex.ml
└── work
    ├── impl.ml
    └── test
        ├── dune

### PR DESCRIPTION
**Pull request by divyankachaudhari**
#2179 

Created 

1. practice/011/ex.ml: it contains the function's signature, which needs to be implemented by the learner;
2. practice/011/work/impl.ml: it includes the template meant to be filed by the learner;
3. practice/011/answer/impl.ml: consists of the solution from the original markdown file.

### CODE REVIEW

File: practice/011/answer/impl.ml
Comment (cuihtlauac) [inline practice/011/answer/impl.ml:1]: Include newline at eof

File: practice/011/ex.ml
Comment (cuihtlauac) [inline practice/011/ex.ml:1]: Type `rle` shouldn't be abstract. Include its definition in the module type. Make sure `dune build --root .` works

**Review by cuihtlauac — CHANGES_REQUESTED** [review]
Thanks @divyankachaudhari, this good. Make sure `dune build --root .` works

File: practice/011/answer/impl.ml
Comment (divyankachaudhari) [reply practice/011/answer/impl.ml:1]: Done

File: practice/011/ex.ml
Comment (divyankachaudhari) [reply practice/011/ex.ml:1]: Include the definition of the `rle` type directly in the Testable signature.

Comment (divyankachaudhari) [reply practice/011/ex.ml:1]: `dune build --root .` works.

**Comment by divyankachaudhari** [comment]
@cuihtlauac Thank you for reviewing. I have made appropriate changes and resolved the comments. `dune build --root .` works 😃

**Comment by divyankachaudhari** [comment]
@cuihtlauac Thank you for reviewing. I have made the appropriate changes and made sure that `dune build --root .` works. Please review it again 😄

**Review by cuihtlauac — CHANGES_REQUESTED** [review]
This is great @divyankachaudhari.
- Add a test case for `encode []` 
- Use local open Tested (see [Modules/Naming and Scoping]https://ocaml.org/docs/modules#naming-and-scoping)

P.S. You don't need to rebase, just add more commits

Thanks

**Comment by divyankachaudhari** [comment]
Hi @cuihtlauac 

> - Add a test case for `encode []`

I have added a test case for empty list `encode []` however this test wasn't mentioned in [original exercise](https://github.com/ocaml/ocaml.org/blob/main/data/exercises/011_modified_run_length.md) (which is why I hadn't added it).

> - Use local open Tested

I have used `open Tested` now. I didn't do it because I wasn't using it _heavily_ as mentioned in the [docs you linked](https://ocaml.org/docs/modules#naming-and-scoping). The folders `001` and `002` don't use `open Tested` either.

Please let me know.

**Review by cuihtlauac — APPROVED** [review]
Great job @divyankachaudhari

**Comment by divyankachaudhari** [comment]
Thank you very much!

**Pull request merged**